File: Phase01/src/Entry.java
package Phase01.src;

class Entry {
    private String docName;
    private int index;
    
    public Entry(String docName, int index) {
        this.docName = docName;
        this.index = index;
    }
    
    public String getdocName() {
        return this.docName;
    }
    
    public int getIndex() {
        return this.index;
    }
    
    @Override
    public String toString() {
        return "Document Id: " + this.docName + ", Index: " + this.index;
    }
    
    @Override
    public boolean equals(Object o) {
        if (!(o instanceof Entry)) {
            return false;
        }
        return this.docName.equals(((Entry) o).docName);
    }
    
    @Override
    public int hashCode() {
        return this.docName.hashCode();
    }
}

File: Phase05/app/library/Entry.cs
namespace library
{
    public class Entry
    {
        public readonly string DocName;
        public readonly int Index;
        public Entry(string docName, int index)
        {
            DocName = docName;
            Index = index;
        }

        public override string ToString()
        {
            return "Document Id: " + this.DocName + ", Index: " + this.Index;
        }

        public override bool Equals(object obj)
        {
            if (obj == null || GetType() != obj.GetType())
            {
                return false;
            }
            return this.DocName.Equals(((Entry)obj).DocName);
        }

        public override int GetHashCode()
        {
            return this.DocName.GetHashCode();
        }
    }
}
